### ISSUE TITLE
Add default Content Type to respond to queries

### ISSUE THREAD

**Issue by alexkreidler** [oxigraph/oxigraph#91]
When I ran a SPARQL query without an `Accept` header, it didn't give a response. I ran the following commands to connect to a 0.2.2 Oxigraph instance running in docker:

```
$ curl -X POST -H 'Content-Type:application/sparql-query' --data 'SELECT * WHERE { ?s ?p ?o } LIMIT 10' http://localhost:7878/query
No suitable ContentEncoding found
$ curl -X POST -H 'Content-Type:application/sparql-query' -H 'Accept: application/sparql-results+json' --data 'SELECT * WHERE { ?s ?p ?o } LIMIT 10' http://localhost:7878/query
{"head":{"vars":["o","p","s"]},"results":{"bindings":[]}}
```
The first query is actually the first bullet point in the [Usage section of the server README](https://github.com/oxigraph/oxigraph/tree/master/server#usage), and the second query is the "Make a query" command under the "Run the web server section".

I think we should provide a default serialization if the user doesn't specify an Accept header. If you want to keep the behavior that it errors without one, it would be great to update that section of docs.

Also, when I made the request with the Accept header set to `text/turtle`, I get the same error. 

Thanks for all your work on this project!

**Comment by Tpt** [comment]
Hi! Sorry for this bug and than you for trying Oxigraph.

I have updated in the latest OXigraph version the content negotiation code to use the one in the HTTP library we use. It seems I did not had enough test coverage and that the new implementation is buggy.

There are actually two distinct problems:
1. The error message is wrong. It should be "No suitable ContentType found" or something similar. I have [submitted a fix to the HTTP library we use](https://github.com/http-rs/http-types/pull/354).
2. There seems to be a content negotiation problem if the header `Accept: */*` is sent. I have reverted the usage of the HTTP library implementation to move back to my custom implementation which should work better.
